scenario_player/utils/token.py

```diff
@@ -15,456 +17,249 @@
     TokenNotDeployed,
     TokenSourceCodeDoesNotExist,
 )
-from scenario_player.services.utils.interface import ServiceInterface
 
 CUSTOM_TOKEN_NAME = "CustomToken"
 
 log = structlog.get_logger(__name__)
 
 
-class Contract:
-    def __init__(self, runner, address=None):
-        self._address = address
-        self.config = runner.definition
-        self._local_rpc_client = runner.client
-        self._local_contract_manager = runner.contract_manager
-        self.interface = ServiceInterface(runner.definition.spaas)
-        self.gas_limit = GAS_LIMIT_FOR_TOKEN_CONTRACT_CALL * 2
-
-    def __repr__(self):
-        return f"<{self.name}>"
-
-    @property
-    def name(self):
-        return f"{self.__class__.__name__}@{to_checksum_address(self.address)}"
-
-    @property
-    def client_id(self):
-        return self.config.spaas.rpc.client_id
-
-    @property
-    def address(self) -> HexAddress:
-        return self._address
-
-    @property
-    def balance(self):
-        return self._local_rpc_client.balance(self.address)
-
-    @property
-    def checksum_address(self) -> ChecksumAddress:
-        """Checksum'd address of the deployed contract."""
-        return to_checksum_address(self.address)
-
-    def transact(self, action: str, parameters: dict) -> TransactionHash:
-        """Send a transact request to `/rpc/contract/<action>` and return the resulting tx hash."""
-        payload = {
-            "client_id": self.client_id,
-            "gas_limit": self.config.gas_limit,
-            "contract_address": self.checksum_address,
-        }
-        payload.update(parameters)
-
-        log.info(f"Requesting '{action}' call", **payload)
-        resp = self.interface.post(f"spaas://rpc/contract/{action}", json=payload)
-        resp.raise_for_status()
-        resp_data = resp.json()
-        tx_hash = resp_data["tx_hash"]
-        log.info(f"'{action}' call succeeded", tx_hash=tx_hash)
-        return tx_hash
-
-    def mint(
```

Comment:
this method is condition, it was moved to a plain function and renamed to `token_maybe_mint`